app/src/server/trpc/routers/service.ts

```diff
@@ -1,9 +1,65 @@
 import { publicProcedure, router } from '../trpc';
-import { createDraft } from '@/server/db/dbOperations';
+import { createDraft, getDraftById, getDraftByUrl } from '@/server/db/dbOperations';
 import { modifyResourceToDraft } from '@/util/modifyResourceFields';
-import { FhirArtifact } from '@/util/types/fhir';
+import { ArtifactResourceType, FhirArtifact } from '@/util/types/fhir';
 import { z } from 'zod';
 import { TRPCError } from '@trpc/server';
+import { DateTime } from 'luxon';
+
+export type ChildArtifactInfo = {
+  resourceType: 'Measure' | 'Library';
+  url: string;
+  version: string;
+};
+
+/**
+ * Helper function that takes an array of related artifacts and recursively
+ * finds all of the child artifacts (related artifacts whose type is composed-of
+ * and has the isOwned extension)
+ */
+async function getChildren(relatedArtifacts: fhir4.RelatedArtifact[]) {
```

Comment:
I suspect that we want to keep this file focused on procedures if we can help it. Can we make a helper or utility file instead and put this function there?

Reply:
Or actually I think maybe it makes sense to make this its own endpoint (query procedure) since it's doing MRS calls to get all of the information.

Reply:
Oh totally- sorry! I didn't the repercussions of that second comment completely through. Ignore and go with helper or utility!